plugins/discourse-narrative-bot/lib/discourse_narrative_bot/new_user_narrative.rb

```diff
@@ -362,6 +378,45 @@ def reply_to_image
       transition ? reply : false
     end
 
+    def missing_likes_like
+      return unless valid_topic?(@post.topic_id)
+      return if @post.user_id == self.discobot_user.id
+
+      fake_delay
+      enqueue_timeout_job(@user)
+
+      last_post = Post.find_by(id: @data[:last_post_id])
+      reply_to(@post, I18n.t("#{I18N_KEY}.likes.not_found", i18n_post_args(url: last_post.url)))
+      false
+    end
+
+    def reply_to_likes
+      post_topic_id = @post.topic_id
+      return unless valid_topic?(post_topic_id)
+
+      post_liked = PostAction.find_by(
```

Comment:
```suggestion
      post_liked = PostAction.exists?(
```

Reply:
Minor suggestion but it seems like we're just checking for the existence here so we don't need to create an ActiveRecord object.